### PR TITLE
ExtrOcamlChar: export ExtrOcamlBasic

### LINKED ISSUE #15076: ExtrOcamlNativeString cannot compile
#### Description of the problem
```coq
From Coq Require Import
     String
     ExtrOcamlNativeString.

Example string_test := string_dec "foo" "bar".
Extraction TestCompile string_test.
```
```ocaml
Error: The implementation /var/folders/d5/f_vqb7sj4cn26rl4cyfxd6f00000gn/T/testextractiondeb0f3.ml
       does not match the interface /var/folders/d5/f_vqb7sj4cn26rl4cyfxd6f00000gn/T/testextractiondeb0f3.cmi:
       Values do not match:
         val string_test : bool/2
       is not included in
         val string_test : sumbool
       File "/var/folders/d5/f_vqb7sj4cn26rl4cyfxd6f00000gn/T/testextractiondeb0f3.mli", line 10, characters 0-25:
         Expected declaration
       File "/var/folders/d5/f_vqb7sj4cn26rl4cyfxd6f00000gn/T/testextractiondeb0f3.ml", line 12, characters 4-15:
         Actual declaration
       File "/var/folders/d5/f_vqb7sj4cn26rl4cyfxd6f00000gn/T/testextractiondeb0f3.ml", lines 2-4, characters 0-7:
         Definition of type bool/1
       File "_none_", line 1:
         Definition of type bool/2
Error:
Compilation of file /var/folders/d5/f_vqb7sj4cn26rl4cyfxd6f00000gn/T/testextractiondeb0f3.ml failed with exit code 2
```
#### Coq Version
```
The Coq Proof Assistant, version 8.15+alpha
compiled with OCaml 4.12.0
```

### PR DESCRIPTION
**Pull request by liyishuai**
Fix #15076

- [x] Added **changelog**.
- [x] Added **test-suite**.

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

**Review by JasonGross — APPROVED** [review]

**Review by herbelin — APPROVED** [review]
Thanks a lot! Would you like to add also the `match x with Empty_string` example?

**Comment by Zimmi48** [comment]
cc @JasonGross: you can probably help debug the coq-tools error here.

**Comment by JasonGross** [comment]
The minimizer cannot minimize coq-tools because coq-tools is too sensitive to the output of coqc (known issue).  From what I can tell, you just got unlucky; there was a half-hour period or so when coq-tools was broken upstream, and I think you just managed to catch that period.  I've restarted the job

**Comment by Zimmi48** [comment]
We should exclude coq-tools from the list of jobs the minimizer proposes to minimize then, either by hardcoding the exclusion in the code of the bot, or by changing the prefix of the job name to something different.

**Comment by JasonGross** [comment]
CI passes now.

@Zimmi48 , I've created https://github.com/coq/bot/pull/176

File: doc/changelog/12-misc/15075-extraction-string.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- **Fixed:**
+  Resolve compilation errors in ExtrOcamlString and ExtrOcamlNativeString
```
Comment (Alizter) [inline doc/changelog/12-misc/15075-extraction-string.rst:2]:
```suggestion
  compilation errors in ExtrOcamlString and ExtrOcamlNativeString
```

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
Excellent, will merge soon.

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**